Objects/genericaliasobject.c

```diff
@@ -275,12 +299,20 @@ subs_tvars(PyObject *obj, PyObject *params, PyObject **argitems)
         }
         for (Py_ssize_t i = 0; i < nsubargs; ++i) {
             PyObject *arg = PyTuple_GET_ITEM(subparams, i);
+            PyObject *subst = arg;
             Py_ssize_t iparam = tuple_index(params, nparams, arg);
             if (iparam >= 0) {
-                arg = argitems[iparam];
+                subst = argitems[iparam];
+            }
+            // convert all the lists inside args to tuples to help
+            // with caching in other libaries if substituting a ParamSpec
+            if (PyList_CheckExact(subst) && is_paramspec(arg)) {
```

Comment:
Why only do this for ParamSpec?

Reply:
I'm afraid of existing code relying on substituting typevars in genericalias with lists, then expecting that list to be inside `__args__`. Although in Python 3.9 that should be invalid, and the IDE/type checker should warn the users about that already.

If I just convert list to tuples for all cases, the C code will become way simpler :).

Reply:
There can't be much code yet relying about anything for GenericAlias -- it's new in 3.9. We could backport that particular behavior to 3.9.3 to ensure there won't be any more code relying on it. :-)